src/assets/js/display-levels.js

```diff
@@ -47,14 +47,24 @@ const DISPLAY_ANY = [
   DISPLAY_LEVELS.authenticated,
   DISPLAY_LEVELS.public,
 ];
+const DISPLAY_PRIVATE_STAFF = [DISPLAY_LEVELS.private, DISPLAY_LEVELS.staff];
 
 const VALID_DISPLAY_LEVELS = {
   primaryUsername: DISPLAY_PUBLIC_ONLY,
   phoneNumbers: DISPLAY_ANY,
-  'accessInformation.ldap': DISPLAY_ANY,
+  'accessInformation.ldap': DISPLAY_PRIVATE_STAFF,
   'accessInformation.mozilliansorg': DISPLAY_NOT_PRIVATE,
 };
 
-export function displayLevelsFor(field) {
-  return (field && VALID_DISPLAY_LEVELS[field]) || DISPLAY_NOT_PRIVATE;
+const NON_STAFF_DISPLAY_LEVELS = {
+  primaryEmail: DISPLAY_ANY,
+};
+
+export function displayLevelsFor(field, scope = null) {
+  return (
+    (field &&
+      ((scope && !scope.isStaff && NON_STAFF_DISPLAY_LEVELS[field]) ||
+        VALID_DISPLAY_LEVELS[field])) ||
+    DISPLAY_NOT_PRIVATE
+  );
```

Comment:
This makes my brain hurt, but making it clearer without making it massively more verbose seems quite difficult. I think we can get rid of some brackets, can't we? And isn't the `field &&` guard unnecessary since `{}[undefined] == undefined`?

I _think_ `scope && !scope.isStaff && NON_STAFF_DISPLAY_LEVELS[field] || VALID_DISPLAY_LEVELS[field] || DISPLAY_NOT_PRIVATE` is logically equivalent.

Reply:
Did that, thanks. Formatter added some parenthesis.